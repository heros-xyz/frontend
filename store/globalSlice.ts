import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  isLoading: false,
  isMute: true,
  isPlaying: false,
  accessToken: '',
  refreshToken: ''
};

export const appState = createSlice({
  name: "appState",
  initialState: initialState,
  reducers: {
    startLoading: (state) => {
      state.isLoading = true;
    },
    finishLoading: (state) => {
      state.isLoading = false;
    },
    playVideo: (state) => {
      state.isPlaying = true;
    },
    pauseVideo: (state) => {
      state.isPlaying = false;
    },
    setAccessToken: (state, action: PayloadAction<string>) => {
      state.accessToken = action.payload
    },
    setRefreshToken: (state, action: PayloadAction<string>) => {
      state.refreshToken = action.payload
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      if (action.payload.appState.accessToken) {
        state.accessToken = action.payload.appState.accessToken;
      }

      if (action.payload.appState.refreshToken) {
        state.refreshToken = action.payload.appState.refreshToken;
      }
    }
  }
});

export const { startLoading, finishLoading, setAccessToken, setRefreshToken } = appState.actions;

export default appState.reducer;
