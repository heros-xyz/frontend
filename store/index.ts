import { configureStore } from "@reduxjs/toolkit";
import { createWrapper } from "next-redux-wrapper";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { setupListeners } from "@reduxjs/toolkit/query";
import globalSlice from "./globalSlice";

export const store = configureStore({
  reducer: {
    appState: globalSlice,

  },
  devTools: true,
  middleware: (gDM) => {
    return gDM().concat(

    );
  },
});

const makeStore = () => store;

export type AppStore = ReturnType<typeof makeStore>;
export type AppState = ReturnType<AppStore["getState"]>;
export type AppDispatch = AppStore["dispatch"];

export const wrapper = createWrapper<AppStore>(makeStore);
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector;

setupListeners(store.dispatch);
