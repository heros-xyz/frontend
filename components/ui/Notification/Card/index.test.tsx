import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "@/store";
import NotificationCard from "./index";

const item = {
  id: "1",
  type: "A_NEW_INTERACTION",
  message: "notification.athlete-new-interaction",
  source: {
    id: "48",
    avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
    fullName: "Sam Smith",
  },
  interaction: {
    id: "70",
    content: "some content should be trimmed if too long",
  },
  comment: null,
  readAt: null,
  createdAt: "2023-02-22T08:43:57.555Z",
};

it("renders homepage unchanged", () => {
  const { container } = render(
    <Provider store={store}>
      <NotificationCard item={item} />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
