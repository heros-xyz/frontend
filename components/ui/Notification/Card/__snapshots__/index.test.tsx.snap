// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <a
    href="/fan/athlete-profile/48/interaction?view=70"
  >
    <div
      class="css-0"
    >
      <div
        class="css-17ztzjy"
      >
        <img
          alt="avatar"
          class="chakra-image css-1fr13bp"
          src="undefined/https://cdn-icons-png.flaticon.com/512/3135/3135715.png"
        />
        <div
          class="css-1bv04n9"
        >
          <span
            class="chakra-heading css-1esm7ln"
          >
            Sam Smith
          </span>
          <span
            class="chakra-text css-19xmqnt"
          >
             has made a new interaction.
          </span>
          <p
            class="chakra-text css-1i719ia"
          >
            4w ago
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
